website/public/js/controllers/groupsCtrl.js

```diff
@@ -224,6 +224,9 @@ habitrpg.controller("GroupsCtrl", ['$scope', '$rootScope', 'Shared', 'Groups', '
       if ($scope.query === undefined || $scope.query === null) {
         return false;
       }
+      if (msg.username == undefined) {
+        return false;
```

Comment:
Usually, I'm very :+1: about being explicit in code 

I did some manual testing with console logs on the social pages, and I think we can be fairly certain that if $scope.query exists, that text exists and is a string. Same goes with msg.user, if it exists at all, it's going to be a string. 

If it ends up causing problems, feel free to open up a new PR.